lib/handler/configurator/errordoc.c

```diff
@@ -42,14 +58,30 @@ static int register_errordoc(h2o_configurator_command_t *cmd, h2o_configurator_c
         if (key->type != YOML_TYPE_SCALAR)
             goto UnknownKeyError;
         if (strcmp(key->data.scalar, "status") == 0) {
-            if (status != -1)
+            if (status_len != 0)
                 goto KeyAlreadyDefinedError;
-            if (h2o_configurator_scanf(cmd, value, "%d", &status) != 0)
-                return -1;
-            if (!(400 <= status && status <= 599)) {
-                h2o_configurator_errprintf(cmd, value, "status must be within range of 400 to 599");
-                return -1;
+
+            if (value->type == YOML_TYPE_SEQUENCE) {
+                if (value->data.sequence.size == 0) {
+                    h2o_configurator_errprintf(cmd, value, "status sequence must not be empty");
+                    return -1;
+                }
+                for (j = 0; j != value->data.sequence.size; ++j) {
+                    if (scan_and_check_status(cmd, value->data.sequence.elements[j], &status[status_len++]) != 0)
```

Comment:
You should either create a temporary value of type `int` for receiving the parsed status code, or, increase the size of the `status` array to 201. Otherwise, an off-by-one error could happen if the YOML array contains something like: 400, 401, 402, 403, ..., 598, 599, 599 (i.e. 200 distinct values followed by any value).